MKSOutputDevice.py

```diff
@@ -403,11 +404,27 @@ def get_max_filename_len(self):
                 return int(global_container_stack.getMetaDataEntry(Constants.MAX_FILENAME_LEN))
         return 30
 
+    def is_auto_file_renaming_enabled(self):
+        global_container_stack = Application.getInstance().getGlobalContainerStack()
+        if global_container_stack:
+            meta_data = global_container_stack.getMetaData()
+            if Constants.AUTO_FILE_RENAMING in meta_data:
+                return bool(global_container_stack.getMetaDataEntry(Constants.AUTO_FILE_RENAMING))
+        return False
+
     def check_valid_filename(self, filename):
-        if filename in self.sdFiles:
-            filename = self.check_valid_filename(self.show_exists_dialog(filename))
-        if len(filename) >= self.get_max_filename_len():
-            filename = self.check_valid_filename(self.show_to_long_dialog(filename))
+        file_already_exists = filename in self.sdFiles
+        max_filename_len = self.get_max_filename_len()
+        filename_too_long = len(filename) > max_filename_len
+
+        if file_already_exists or filename_too_long:
+            if self.is_auto_file_renaming_enabled():
+                filename = utils.generate_new_filename(self.sdFiles, filename, max_filename_len)
```

Comment:
Good catch!
In case of ".gcode" in the input it wasn't working as expected, it was generating ".gcode1", ".gcode2" instead of "1.gcode", "2.gcode". 
Added new test for this **test_generate_new_filename_for_file_with_only_extension** and had to edit **utils.py** and implement custom **split_extension** method